connectors/sources/tests/test_mssql.py

```diff
@@ -0,0 +1,39 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Tests the microsoft sql database source class methods"""
+from unittest.mock import Mock, patch
+
+import pytest
+
+from connectors.sources.mssql import MSSQLDataSource
+from connectors.sources.tests.support import create_source
+from connectors.tests.commons import AsyncIterator
+
+
+@pytest.mark.asyncio
+async def test_ping(patch_logger):
+    """Test ping method of MSSQLDataSource class"""
+    # Setup
+    source = create_source(MSSQLDataSource)
+    source._create_engine = Mock()
+    source.execute_query = Mock(return_value=AsyncIterator(["table1", "table2"]))
+
+    # Execute
+    await source.ping()
```

Comment:
As we're inheriting this from `generic_database.py` and we're mocking the real database calls it would make sense to lift these test cases to `test_generic_database.py` without reimplementing them for every data source inheriting from `generic_database.py`.